### PR TITLE
Merging bookmarks to main

### PR DESCRIPTION
**Pull request by soni2261**
This PR introduces episode bookmarks implementation to the frontend, it allows users to add a bookmark to a specific time when listening to an episode.

Features:
Adding a bookmark
Deleting a bookmark
Getting a list of all bookmarks

Implementation:
Created Bookmarks component and UI Module
Added POST, GET and DELETE Requests to Bookmarks Component using Bookmarks Helper
Created Request and Response Interfaces for Bookmarks

closes https://github.com/awaazo/awaazo/issues/166

### CODE REVIEW

**Comment by soni2261** [comment]
Last PR wouldn't work so I created this merge request, it should all work fine. The UI will still be worked on but I need to pull in order to access the NowPlaying page

**Review by mathieutrudeau — APPROVED** [review]
Looks good. The file naming seems to be inconsistent tho. bookmarkComponent.tsx should probably be named BookmarkComponent.tsx

File: frontend/components/bookmarks/bookmarkComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+const BookmarkComponent = ({ episodeId, selectedTimestamp}) => {
+  // Component Values
+  const [bookmarks, setBookmarks] = useState<Bookmark[]>([]);
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const isMobile = useBreakpointValue({ base: true, md: false });
+  const [newTitle, setNewTitle] = useState("");
+  const [newNote, setNewNote] = useState("");
+
+  // Fetch episode details and transform bookmarks
+  // useEffect(() => {
```
Comment (jonathan-githubofficial) [inline frontend/components/bookmarks/bookmarkComponent.tsx]: pls remove any commented out code that won't be used in the future

File: frontend/components/bookmarks/bookmarkComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,237 @@
+import { useEffect, useState } from "react";
```
Comment (jonathan-githubofficial) [inline frontend/components/bookmarks/bookmarkComponent.tsx]: bookmarkComponent.tsx should have a more descriptive name like :  CreateBookmarkModule

**Review by jonathan-githubofficial — APPROVED** [review]

**Pull request merged**